processing/src/main/java/org/apache/druid/query/aggregation/post/FinalizingFieldAccessPostAggregator.java

```diff
@@ -94,29 +104,35 @@ public String getName()
     return name;
   }
 
+  @Override
+  public ValueType getType()
+  {
+    return finalizedType;
+  }
+
   @Override
   public FinalizingFieldAccessPostAggregator decorate(final Map<String, AggregatorFactory> aggregators)
   {
     final Comparator<Object> theComparator;
     final Function<Object, Object> theFinalizer;
+    final ValueType finalizedType;
 
     if (aggregators != null && aggregators.containsKey(fieldName)) {
       //noinspection unchecked
       theComparator = aggregators.get(fieldName).getComparator();
+      theFinalizer = aggregators.get(fieldName)::finalizeComputation;
+      finalizedType = aggregators.get(fieldName).getFinalizedType();
     } else {
       //noinspection unchecked
       theComparator = (Comparator) Comparators.naturalNullsFirst();
-    }
-
-    if (aggregators != null && aggregators.containsKey(fieldName)) {
-      theFinalizer = aggregators.get(fieldName)::finalizeComputation;
-    } else {
       theFinalizer = Function.identity();
+      finalizedType = ValueTypes.defaultAggregationType();
```

Comment:
Similar comment to FieldAccessPostAggregator: why not keep it null here? What are the pros and cons of null vs `ValueTypes.defaultAggregationType()`?